### PR TITLE
Add HTTP timeout support for pinot query

### PR DESCRIPTION
**Pull request by zaheersyedkdr**
1. Currently Pinot uses default golang HTTP timeout, which is "No TImeout". Due to this, users of pinot go client have to implement timeout logic on their side, to avoid waiting for pinot query.
2. This PR aims to solve this by exposing HTTP Timeout parameter to the end user.

### CODE REVIEW

**Comment by zaheersyedkdr** [comment]
@xiangfu0 , @Jackie-Jiang for PR review ^

**Comment by zaheersyedkdr** [comment]
Gentle reminder to review

**Review by xiangfu0 — APPROVED** [review]

File: pinot/connectionFactory_test.go [hunk 1]
```diff
@@ -35,3 +36,17 @@ func TestPinotClients(t *testing.T) {
...
+		BrokerList: []string{"www.google.com:81"},
+		HTTPTimeout: 1 * time.Second,
+	})
+	assert.Nil(t,err)
+	start := time.Now()
+	_ , err = pinotClient.ExecuteSQL("testTable","select * from testTable");
+	end := time.Since(start)
+	assert.NotNil(t,err)
+	diff := int(end.Seconds())
+	assert.Equal(t,1,diff)
```
Comment (xiangfu0) [inline pinot/connectionFactory_test.go:51]: How accurate is this time check? Shall we make a range check like smaller than 2?

**Review by xiangfu0 — APPROVED** [review]
Overall lgtm

File: pinot/connectionFactory_test.go (hunk 1 above)
Comment (zaheersyedkdr) [reply pinot/connectionFactory_test.go]: Updated. Ideally "int" conversion should round it off. But for safer side, I added a less than 2 sec assertion

**Comment by xiangfu0** [comment]
Many thanks for your contribution!

**Pull request merged**